### ISSUE TITLE
[REVIEW]: Explaining Black-Box Models through Counterfactuals

### ISSUE THREAD

**Issue by whedon** [JuliaCon/proceedings-review#130]
**Submitting author:** @pat-alt (<a href="http://orcid.org/0000-0003-4726-8613">Patrick Altmeyer</a>)
**Repository:** <a href="https://github.com/pat-alt/CounterfactualExplanations.jl" target ="_blank">https://github.com/pat-alt/CounterfactualExplanations.jl</a>
**Version:** v0.1.14
**Reviewers:** @kartikeyrinwa
**Archive:** <a href="https://doi.org/10.5281/zenodo.8239379" target="_blank">10.5281/zenodo.8239379</a>

## Status

[![status](https://proceedings.juliacon.org/papers/f0aaf97980595c2ade2f4d17601bd5e0/status.svg)](https://proceedings.juliacon.org/papers/f0aaf97980595c2ade2f4d17601bd5e0)

Status badge code:

```
HTML: <a href="https://proceedings.juliacon.org/papers/f0aaf97980595c2ade2f4d17601bd5e0"><img src="https://proceedings.juliacon.org/papers/f0aaf97980595c2ade2f4d17601bd5e0/status.svg"></a>
Markdown: [![status](https://proceedings.juliacon.org/papers/f0aaf97980595c2ade2f4d17601bd5e0/status.svg)](https://proceedings.juliacon.org/papers/f0aaf97980595c2ade2f4d17601bd5e0)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/pat-alt/CounterfactualExplanations.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@jmaces & @kartikeyrinwa, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @matbesancon know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pat-alt/CounterfactualExplanations.jl">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Authorship:** Has the submitting author (@pat-alt) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [ ] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [ ] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [ ] **Context:** is the scientific context motivating the work correctly presented?
- [ ] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [ ] **Results:** are the results presented and compared to approaches with similar goals?
### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pat-alt/CounterfactualExplanations.jl">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Authorship:** Has the submitting author (@pat-alt) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [ ] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [ ] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [ ] **Context:** is the scientific context motivating the work correctly presented?
- [ ] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [ ] **Results:** are the results presented and compared to approaches with similar goals?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @jmaces, @kartikeyrinwa it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/JuliaCon/proceedings-review) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/JuliaCon/proceedings-review:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by matbesancon** [comment]
@whedon remove @jmaces as reviewer

**Comment by whedon** [comment]
Failed to discover a `Statement of need` section in paper

**Comment by whedon** [comment]
Wordcount for `paper.tex` is 6143

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00130/jcon.00130/10.21105.jcon.00130.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00130/jcon.00130/10.21105.jcon.00130.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
OK, @jmaces is no longer a reviewer

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=1.16 s (170.3 files/s, 100136.5 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
SVG                             39              0              0          91627
TeX                             11           1010            195           8336
Julia                           59            776            286           4445
HTML                             1            257              2           3044
Markdown                        33            806              0           2094
CSS                              4             36             12           1838
JavaScript                       6             80             88            633
TOML                             8             42              0            303
JSON                            28              0              0            281
YAML                             8              5              2            173
Ruby                             1              8              4             45
-------------------------------------------------------------------------------
SUM:                           198           3020            589         112819
-------------------------------------------------------------------------------


Statistical information for the repository '7d188eca19badabba11bc5f2' was
gathered on 2023/06/20.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
pat-alt                         15          5731           4186          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
pat-alt                    1659           28.9          0.8               10.85
```

**Comment by matbesancon** [comment]
@pat-alt can you check the review posted here: https://github.com/JuliaCon/proceedings-review/issues/103#issuecomment-1598649885
and edit accordingly? Let us know when it is complete

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.02704 is OK
- doi.org/10.1038/s41597-020-0448-0 is OK
- 10.1214/20-AOS1965 is OK
- 10.48550/arXiv.1810.08810 is OK

MISSING DOIs

- 10.1109/satml54575.2023.00036 may be a valid DOI for title: Endogenous Macrodynamics in Algorithmic Recourse
- 10.2139/ssrn.195733 may be a valid DOI for title: Instrumental Variables Estimates of the Effect of Subsidized Training on the Quantiles of Trainee Earnings
- 10.1109/deeptest52559.2021.00007 may be a valid DOI for title: Machine Learning Model Drift Detection Via Weak Data Slices
- 10.3386/w20343 may be a valid DOI for title: Reference-Dependent Preferences: Evidence from Marathon Runners
- 10.1257/aer.99.1.486 may be a valid DOI for title: Indirect Effects of an Aid Program: How Do Cash Transfers Affect Ineligibles’ Consumption?
- 10.1017/cbo9780511618031.014 may be a valid DOI for title: “Coherent Arbitrariness”: Stable Demand Curves without Stable Preferences
- 10.1017/cbo9780511618031.015 may be a valid DOI for title: Tom Sawyer and the Construction of Value
- 10.1016/j.inffus.2019.12.012 may be a valid DOI for title: Explainable Artificial Intelligence (XAI): Concepts, Taxonomies, Opportunities and Challenges toward Responsible AI
- 10.7551/mitpress/3077.003.0044 may be a valid DOI for title: Deciding Advantageously before Knowing the Advantageous Strategy
- 10.1007/978-1-4419-9096-9 may be a valid DOI for title: Reproducing Kernel Hilbert Spaces in Probability and Statistics
- 10.3386/w3487 may be a valid DOI for title: The Federal Funds Rate and the Channels of Monetary Transnission
- 10.1016/j.techsoc.2021.101852 may be a valid DOI for title: Machine Learning, Knowledge Risk, and Principal-Agent Problems in Automated Trading
- 10.2139/ssrn.965818 may be a valid DOI for title: Identification of Peer Effects through Social Networks
- 10.2139/ssrn.3534495 may be a valid DOI for title: Peer Effects in Networks: A Survey
- 10.2307/2234554 may be a valid DOI for title: Nonlinear Dynamics, Chaos, and Instability: Statistical Theory and Economic Evidence
- 10.3386/w4509 may be a valid DOI for title: Minimum Wages and Employment: A Case Study of the Fast Food Industry in New Jersey and Pennsylvania
- 10.1109/sp.2017.49 may be a valid DOI for title: Towards Evaluating the Robustness of Neural Networks
- 10.3386/w14032 may be a valid DOI for title: Does Your Cohort Matter? Measuring Peer Effects in College Achievement
- 10.3982/ecta10168 may be a valid DOI for title: From Natural Variation to Optimal Policy? The Importance of Endogenous Peer Group Formation
- 10.2139/ssrn.4090707 may be a valid DOI for title: Explainable Artificial Intelligence: Interpreting Default Forecasting Models Based on Machine Learning
- 10.3386/w15617 may be a valid DOI for title: Adjustment Costs, Firm Responses, and Micro vs. Macro Labor Supply Elasticities: Evidence from Danish Tax Records
- 10.1214/18-aoas1222 may be a valid DOI for title: Variable Prioritization in Nonlinear Black Box Methods: A Genetic Association Case Study
- 10.1145/3514094.3534144 may be a valid DOI for title: Counterfactual Explanations for Prediction and Diagnosis in Xai
- 10.1080/01621459.1999.10473858 may be a valid DOI for title: Causal Effects in Nonexperimental Studies: Reevaluating the Evaluation of Training Programs
- 10.2139/ssrn.1596425 may be a valid DOI for title: The Persistent Effects of Peru’s Mining Mita
- 10.3233/ds-200028 may be a valid DOI for title: Reinforcement Learning for Personalization: A Systematic Literature Review
- 10.3390/app11167274 may be a valid DOI for title: A Framework and Benchmarking Study for Counterfactual Generating Methods on Tabular Data
- 10.1037/0022-3514.37.7.1097 may be a valid DOI for title: The Stability of Behavior: I. On Predicting Most of the People Much of the Time.
- 10.1086/497818 may be a valid DOI for title: Clean Evidence on Peer Effects
- 10.2139/ssrn.203194 may be a valid DOI for title: Cooperation and Punishment in Public Goods Experiments
- 10.1287/mnsc.2017.2908 may be a valid DOI for title: On the External Validity of Social Preference Games: A Systematic Lab-Field Study
- 10.1037/0022-3514.75.3.617 may be a valid DOI for title: Immune Neglect: A Source of Durability Bias in Affective Forecasting.
- 10.1037/e722842011-016 may be a valid DOI for title: The Uncertainty Effect: When a Risky Prospect Is Valued Less than Its Worst Possible Outcome
- 10.1007/978-3-030-42553-1_3 may be a valid DOI for title: Bayesian Neural Networks: An Introduction and Survey
- 10.3386/w11562 may be a valid DOI for title: The Incredible Volcker Disinflation
- 10.1920/wp.cem.2017.0817 may be a valid DOI for title: An Econometric Model of Network Formation with Degree Heterogeneity
- 10.1017/cbo9780511618031.006 may be a valid DOI for title: Economic Theory of Choice and the Preference Reversal Phenomenon
- 10.1086/717935 may be a valid DOI for title: Name Your Friends, but Only Five? The Importance of Censoring in Peer Effects Estimates Using Social Network Data
- 10.1109/icmla.2011.144 may be a valid DOI for title: Thompson Sampling for Dynamic Multi-Armed Bandits
- 10.1016/j.ins.2008.07.024 may be a valid DOI for title: Improving Artificial Neural Networks’ Performance in Seasonal Time Series Forecasting
- 10.1145/1273496.1273541 may be a valid DOI for title: A Bound on the Label Complexity of Agnostic Active Learning
- 10.1177/2053951720926558 may be a valid DOI for title: The Virtue of Simplicity: On Machine Learning Models in Algorithmic Trading
- 10.1509/jmkr.48.spl.s23 may be a valid DOI for title: Increasing Saving Behavior through Age-Progressed Renderings of the Future Self
- 10.1007/978-1-4899-7687-1_100391 may be a valid DOI for title: Random Decision Forests
- 10.3150/22-bej1484 may be a valid DOI for title: Bayes-Optimal Prediction with Frequentist Coverage Control
- 10.1006/obhd.1996.0077 may be a valid DOI for title: The Evaluability Hypothesis: An Explanation for Preference Reversals between Joint and Separate Evaluations of Alternatives
- 10.1017/cbo9780511618031.033 may be a valid DOI for title: Music, Pandas, and Muggers: On the Affective Psychology of Value.
- 10.1002/jae.2426 may be a valid DOI for title: A Social Interactions Model with Endogenous Friendship Formation and Selectivity
- 10.21105/joss.00602 may be a valid DOI for title: Flux: Elegant Machine Learning with Julia
- 10.1257/aer.97.3.890 may be a valid DOI for title: Meeting Strangers and Friends of Friends: How Random Are Social Networks?
- 10.1126/science.1111709 may be a valid DOI for title: Failure to Detect Mismatches between Intention and Outcome in a Simple Decision Task
- 10.2139/ssrn.3043404 may be a valid DOI for title: Estimation of Peer Effects in Endogenous Social Networks: Control Function Approach
- 10.1198/1061860032148 may be a valid DOI for title: A Modified Principal Component Technique Based on the LASSO
- 10.1109/mci.2022.3155327 may be a valid DOI for title: Hands-on Bayesian Neural Networks–a Tutorial for Deep Learning Users
- 10.1017/cbo9781139175197.009 may be a valid DOI for title: Experimental Tests of the Endowment Effect and the Coase Theorem
- 10.1016/0749-5978(92)90015-y may be a valid DOI for title: Reference Points, Anchors, Norms, and Mixed Feelings
- 10.1145/3313831.3376219 may be a valid DOI for title: Interpreting Interpretability: Understanding Data Scientists’ Use of Interpretability Tools for Machine Learning
- 10.4314/ajst.v5i2.15330 may be a valid DOI for title: Seasonal Time Series Forecasting: A Comparative Study of ARIMA and ANN Models
- 10.1007/978-3-030-93842-0_6 may be a valid DOI for title: Exploring Explainable AI in the Financial Sector: Perspectives of Banks and Supervisory Authorities
- 10.1081/etc-120025891 may be a valid DOI for title: Best Spatial Two-Stage Least Squares Estimators for a Spatial Autoregressive Model with Autoregressive Disturbances
- 10.1177/0956797612450302 may be a valid DOI for title: The Financial Costs of Sadness
- 10.3386/w9736 may be a valid DOI for title: Neoclassical Theory versus Prospect Theory: Evidence from the Marketplace
- 10.3386/w7682 may be a valid DOI for title: The Power of Suggestion: Inertia in 401 (k) Participation and Savings Behavior
- 10.2307/2298123 may be a valid DOI for title: Identification of Endogenous Social Effects: The Reflection Problem
- 10.2139/ssrn.2523510 may be a valid DOI for title: Goals as Reference Points in Marathon Running: A Novel Test of Reference Dependence
- 10.20955/wp.2015.012 may be a valid DOI for title: FRED-MD: A Monthly Database for Macroeconomic Research
- 10.1007/s10618-013-0342-x may be a valid DOI for title: Visualizing Multi-Dimensional Decision Boundaries in 2D
- 10.1016/j.artint.2018.07.007 may be a valid DOI for title: Explanation in Artificial Intelligence: Insights from the Social Sciences
- 10.1037/0022-3514.54.4.687 may be a valid DOI for title: The Nature of Adolescent Competencies Predicted by Preschool Delay of Gratification.
- 10.1145/3287560.3287574 may be a valid DOI for title: Explaining Explanations in AI
- 10.1145/3351095.3372850 may be a valid DOI for title: Explaining Machine Learning Classifiers through Diverse Counterfactual Explanations
- 10.1109/cisda.2015.7208643 may be a valid DOI for title: Evaluating Model Drift in Machine Learning Algorithms
- 10.1016/s0167-7152(96)00140-x may be a valid DOI for title: Sparse Spatial Autoregressions
- 10.1145/3241036 may be a valid DOI for title: The Seven Tools of Causal Inference, with Reflections on Machine Learning
- 10.5860/choice.38-4002 may be a valid DOI for title: Economic Events, Ideas, and Policies: The 1960s and After
- 10.1016/j.jeconom.2014.08.008 may be a valid DOI for title: Estimating a Spatial Autoregressive Model with an Endogenous Spatial Weight Matrix
- 10.3386/w2966 may be a valid DOI for title: Does Monetary Policy Matter? A New Test in the Spirit of Friedman and Schwartz
- 10.1038/s42256-019-0048-x may be a valid DOI for title: Stop Explaining Black Box Machine Learning Models for High Stakes Decisions and Use Interpretable Models Instead
- 10.3386/w7469 may be a valid DOI for title: Peer Effects with Random Assignment: Results for Dartmouth Roommates
- 10.1080/01621459.2017.1395341 may be a valid DOI for title: Least Ambiguous Set-Valued Classifiers with Bounded Error Levels
- 10.1109/icdcsw.2011.20 may be a valid DOI for title: Finding a" Kneedle" in a Haystack: Detecting Knee Points in System Behavior
- 10.1086/209205 may be a valid DOI for title: Choice Based on Reasons: The Case of Attraction and Compromise Effects
- 10.21034/qr.1011 may be a valid DOI for title: Are Forecasting Models Usable for Policy Analysis?
- 10.1037/e301532005-001 may be a valid DOI for title: Who Accepts Savage’s Axiom?
- 10.1109/tmm.2014.2330697 may be a valid DOI for title: A Simple Method to Determine If a Music Information Retrieval System Is a “Horse”
- 10.1017/cbo9780511618031.039 may be a valid DOI for title: Libertarian Paternalism Is Not an Oxymoron
- 10.1016/0165-1765(81)90067-7 may be a valid DOI for title: Some Empirical Evidence on Dynamic Inconsistency
- 10.1086/380085 may be a valid DOI for title: Save More Tomorrow™: Using Behavioral Economics to Increase Employee Saving
- 10.1007/978-1-4613-2391-4_2 may be a valid DOI for title: The Framing of Decisions and the Psychology of Choice
- 10.1177/0956797610396225 may be a valid DOI for title: How Incidental Values from the Environment Affect Decisions about Money, Risk, and Delay
- 10.1145/3287560.3287566 may be a valid DOI for title: Actionable Recourse in Linear Classification
- 10.1037/0022-3514.79.1.66 may be a valid DOI for title: Egocentric Empathy Gaps between Owners and Buyers: Misperceptions of the Endowment Effect.
- 10.2139/ssrn.3589337 may be a valid DOI for title: Modeling Multivariate Time Series in Economics: From Auto-Regressions to Recurrent Neural Networks
- 10.2139/ssrn.3063289 may be a valid DOI for title: Counterfactual Explanations without Opening the Black Box: Automated Decisions and the GDPR
- 10.1007/bf00116900 may be a valid DOI for title: Learning in the Presence of Concept Drift and Hidden Contexts
- 10.1093/biostatistics/kxp008 may be a valid DOI for title: A Penalized Matrix Decomposition, with Applications to Sparse Principal Components and Canonical Correlation Analysis
- 10.1016/j.eswa.2007.12.020 may be a valid DOI for title: The Comparisons of Data Mining Techniques for the Predictive Accuracy of Probability of Default of Credit Card Clients
- 10.1016/s0925-2312(01)00702-0 may be a valid DOI for title: Time Series Forecasting Using a Hybrid ARIMA and Neural Network Model

INVALID DOIs

- None
```

**Comment by pat-alt** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00130/jcon.00130/10.21105.jcon.00130.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00130/jcon.00130/10.21105.jcon.00130.pdf) :page_facing_up: :point_left:

**Comment by pat-alt** [comment]
@matbesancon @kartikeyrinwa I've now edited according to https://github.com/JuliaCon/proceedings-review/issues/103#issuecomment-1598649885. 

In addition to the helpful suggestions, I have incorporated recent updates to the package including:

- [Composability of Generators](https://juliatrustworthyai.github.io/CounterfactualExplanations.jl/dev/tutorials/generators/)
- [Evaluation and Benchmarking](https://juliatrustworthyai.github.io/CounterfactualExplanations.jl/dev/tutorials/evaluation/)
- Contributions that have come out of the [TU Delft Student Software Project](https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/pull/252)

I noted that the boxes for *testing* and *contribution guidelines* in the Documentation section were unticked. We do run tests, of course - should that be explained in the docs? As for contribution guidelines, we have a section in the [docs](https://juliatrustworthyai.github.io/CounterfactualExplanations.jl/dev/contribute/) and the README - should we improve this somehow? 

Thanks again!

**Comment by kartikeyrinwa** [comment]
### Review

As the authors have sufficiently address my previous comments, **I recommend acceptance of the paper**.

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pat-alt/CounterfactualExplanations.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@pat-alt) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?

**Comment by matbesancon** [comment]
@whedon accept

**Comment by whedon** [comment]
To recommend a paper to be accepted use `@whedon recommend-accept`

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by matbesancon** [comment]
@whedon set v0.1.13 as version

**Comment by whedon** [comment]
OK. v0.1.13 is the version.

**Comment by matbesancon** [comment]
@pat-alt the last step we will need will be making an archive on Zenodo of the code on the repository, and writing the DOI here

**Comment by pat-alt** [comment]
@matbesancon awesome, thanks. That's all done: [![DOI](https://zenodo.org/badge/440782065.svg)](https://zenodo.org/badge/latestdoi/440782065) 

Had to push a new version  `v0.1.14` for this to take effect.

**Comment by matbesancon** [comment]
@whedon set v0.1.14 as version

**Comment by matbesancon** [comment]
@whedon set 10.5281/zenodo.8239379 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.8239379" target="_blank">10.5281/zenodo.8239379</a> is the archive.

**Comment by matbesancon** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @JuliaCon/jcon-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/JuliaCon/proceedings-papers/pull/67

If the paper PDF and Crossref deposit XML look good in https://github.com/JuliaCon/proceedings-papers/pull/67, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.48550/arXiv.1412.6980 is OK
- 10.48550/arXiv.1708.07747 is OK
- 10.21105/joss.02704 is OK
- doi.org/10.1038/s41597-020-0448-0 is OK
- 10.1214/20-AOS1965 is OK
- 10.48550/arXiv.1810.08810 is OK
- 10.1007/s10618-022-00831-6 is OK
- 10.48550/arXiv.1912.03277 is OK
- 10.1093/pan/mpq009 is OK
- 10.48550/arXiv.1903.08689 is OK
- 10.24432/C5XW20 is OK
- 10.1145/3097983.3098039 is OK
- 10.48550/arXiv.1712.08443 is OK

MISSING DOIs

- 10.1093/gmo/9781561592630.article.j990050 may be a valid DOI for title: "Fidelity"
- 10.1109/satml54575.2023.00036 may be a valid DOI for title: Endogenous Macrodynamics in Algorithmic Recourse
- 10.2139/ssrn.195733 may be a valid DOI for title: Instrumental Variables Estimates of the Effect of Subsidized Training on the Quantiles of Trainee Earnings
- 10.1109/deeptest52559.2021.00007 may be a valid DOI for title: Machine Learning Model Drift Detection Via Weak Data Slices
- 10.3386/w20343 may be a valid DOI for title: Reference-Dependent Preferences: Evidence from Marathon Runners
- 10.1257/aer.99.1.486 may be a valid DOI for title: Indirect Effects of an Aid Program: How Do Cash Transfers Affect Ineligibles’ Consumption?
- 10.1017/cbo9780511618031.014 may be a valid DOI for title: “Coherent Arbitrariness”: Stable Demand Curves without Stable Preferences
- 10.1017/cbo9780511618031.015 may be a valid DOI for title: Tom Sawyer and the Construction of Value
- 10.1016/j.inffus.2019.12.012 may be a valid DOI for title: Explainable Artificial Intelligence (XAI): Concepts, Taxonomies, Opportunities and Challenges toward Responsible AI
- 10.7551/mitpress/3077.003.0044 may be a valid DOI for title: Deciding Advantageously before Knowing the Advantageous Strategy
- 10.1007/978-1-4419-9096-9 may be a valid DOI for title: Reproducing Kernel Hilbert Spaces in Probability and Statistics
- 10.3386/w3487 may be a valid DOI for title: The Federal Funds Rate and the Channels of Monetary Transnission
- 10.1016/j.techsoc.2021.101852 may be a valid DOI for title: Machine Learning, Knowledge Risk, and Principal-Agent Problems in Automated Trading
- 10.2139/ssrn.965818 may be a valid DOI for title: Identification of Peer Effects through Social Networks
- 10.2139/ssrn.3534495 may be a valid DOI for title: Peer Effects in Networks: A Survey
- 10.2307/2234554 may be a valid DOI for title: Nonlinear Dynamics, Chaos, and Instability: Statistical Theory and Economic Evidence
- 10.3386/w4509 may be a valid DOI for title: Minimum Wages and Employment: A Case Study of the Fast Food Industry in New Jersey and Pennsylvania
- 10.1109/sp.2017.49 may be a valid DOI for title: Towards Evaluating the Robustness of Neural Networks
- 10.3386/w14032 may be a valid DOI for title: Does Your Cohort Matter? Measuring Peer Effects in College Achievement
- 10.3982/ecta10168 may be a valid DOI for title: From Natural Variation to Optimal Policy? The Importance of Endogenous Peer Group Formation
- 10.2139/ssrn.4090707 may be a valid DOI for title: Explainable Artificial Intelligence: Interpreting Default Forecasting Models Based on Machine Learning
- 10.3386/w15617 may be a valid DOI for title: Adjustment Costs, Firm Responses, and Micro vs. Macro Labor Supply Elasticities: Evidence from Danish Tax Records
- 10.1214/18-aoas1222 may be a valid DOI for title: Variable Prioritization in Nonlinear Black Box Methods: A Genetic Association Case Study
- 10.1145/3514094.3534144 may be a valid DOI for title: Counterfactual Explanations for Prediction and Diagnosis in Xai
- 10.1080/01621459.1999.10473858 may be a valid DOI for title: Causal Effects in Nonexperimental Studies: Reevaluating the Evaluation of Training Programs
- 10.2139/ssrn.1596425 may be a valid DOI for title: The Persistent Effects of Peru’s Mining Mita
- 10.3233/ds-200028 may be a valid DOI for title: Reinforcement Learning for Personalization: A Systematic Literature Review
- 10.3390/app11167274 may be a valid DOI for title: A Framework and Benchmarking Study for Counterfactual Generating Methods on Tabular Data
- 10.1037/0022-3514.37.7.1097 may be a valid DOI for title: The Stability of Behavior: I. On Predicting Most of the People Much of the Time.
- 10.1086/497818 may be a valid DOI for title: Clean Evidence on Peer Effects
- 10.2139/ssrn.203194 may be a valid DOI for title: Cooperation and Punishment in Public Goods Experiments
- 10.1287/mnsc.2017.2908 may be a valid DOI for title: On the External Validity of Social Preference Games: A Systematic Lab-Field Study
- 10.1037/0022-3514.75.3.617 may be a valid DOI for title: Immune Neglect: A Source of Durability Bias in Affective Forecasting.
- 10.1037/e722842011-016 may be a valid DOI for title: The Uncertainty Effect: When a Risky Prospect Is Valued Less than Its Worst Possible Outcome
- 10.1007/978-3-030-42553-1_3 may be a valid DOI for title: Bayesian Neural Networks: An Introduction and Survey
- 10.3386/w11562 may be a valid DOI for title: The Incredible Volcker Disinflation
- 10.1920/wp.cem.2017.0817 may be a valid DOI for title: An Econometric Model of Network Formation with Degree Heterogeneity
- 10.1017/cbo9780511618031.006 may be a valid DOI for title: Economic Theory of Choice and the Preference Reversal Phenomenon
- 10.1086/717935 may be a valid DOI for title: Name Your Friends, but Only Five? The Importance of Censoring in Peer Effects Estimates Using Social Network Data
- 10.1109/icmla.2011.144 may be a valid DOI for title: Thompson Sampling for Dynamic Multi-Armed Bandits
- 10.1016/j.ins.2008.07.024 may be a valid DOI for title: Improving Artificial Neural Networks’ Performance in Seasonal Time Series Forecasting
- 10.1145/1273496.1273541 may be a valid DOI for title: A Bound on the Label Complexity of Agnostic Active Learning
- 10.1177/2053951720926558 may be a valid DOI for title: The Virtue of Simplicity: On Machine Learning Models in Algorithmic Trading
- 10.1509/jmkr.48.spl.s23 may be a valid DOI for title: Increasing Saving Behavior through Age-Progressed Renderings of the Future Self
- 10.1007/978-1-4899-7687-1_100391 may be a valid DOI for title: Random Decision Forests
- 10.3150/22-bej1484 may be a valid DOI for title: Bayes-Optimal Prediction with Frequentist Coverage Control
- 10.1006/obhd.1996.0077 may be a valid DOI for title: The Evaluability Hypothesis: An Explanation for Preference Reversals between Joint and Separate Evaluations of Alternatives
- 10.1017/cbo9780511618031.033 may be a valid DOI for title: Music, Pandas, and Muggers: On the Affective Psychology of Value.
- 10.1002/jae.2426 may be a valid DOI for title: A Social Interactions Model with Endogenous Friendship Formation and Selectivity
- 10.21105/joss.00602 may be a valid DOI for title: Flux: Elegant Machine Learning with Julia
- 10.1257/aer.97.3.890 may be a valid DOI for title: Meeting Strangers and Friends of Friends: How Random Are Social Networks?
- 10.1126/science.1111709 may be a valid DOI for title: Failure to Detect Mismatches between Intention and Outcome in a Simple Decision Task
- 10.2139/ssrn.3043404 may be a valid DOI for title: Estimation of Peer Effects in Endogenous Social Networks: Control Function Approach
- 10.1198/1061860032148 may be a valid DOI for title: A Modified Principal Component Technique Based on the LASSO
- 10.1109/mci.2022.3155327 may be a valid DOI for title: Hands-on Bayesian Neural Networks–a Tutorial for Deep Learning Users
- 10.1017/cbo9781139175197.009 may be a valid DOI for title: Experimental Tests of the Endowment Effect and the Coase Theorem
- 10.1016/0749-5978(92)90015-y may be a valid DOI for title: Reference Points, Anchors, Norms, and Mixed Feelings
- 10.1145/3313831.3376219 may be a valid DOI for title: Interpreting Interpretability: Understanding Data Scientists’ Use of Interpretability Tools for Machine Learning
- 10.4314/ajst.v5i2.15330 may be a valid DOI for title: Seasonal Time Series Forecasting: A Comparative Study of ARIMA and ANN Models
- 10.1007/978-3-030-93842-0_6 may be a valid DOI for title: Exploring Explainable AI in the Financial Sector: Perspectives of Banks and Supervisory Authorities
- 10.1081/etc-120025891 may be a valid DOI for title: Best Spatial Two-Stage Least Squares Estimators for a Spatial Autoregressive Model with Autoregressive Disturbances
- 10.1177/0956797612450302 may be a valid DOI for title: The Financial Costs of Sadness
- 10.3386/w9736 may be a valid DOI for title: Neoclassical Theory versus Prospect Theory: Evidence from the Marketplace
- 10.3386/w7682 may be a valid DOI for title: The Power of Suggestion: Inertia in 401 (k) Participation and Savings Behavior
- 10.2307/2298123 may be a valid DOI for title: Identification of Endogenous Social Effects: The Reflection Problem
- 10.2139/ssrn.2523510 may be a valid DOI for title: Goals as Reference Points in Marathon Running: A Novel Test of Reference Dependence
- 10.20955/wp.2015.012 may be a valid DOI for title: FRED-MD: A Monthly Database for Macroeconomic Research
- 10.1007/s10618-013-0342-x may be a valid DOI for title: Visualizing Multi-Dimensional Decision Boundaries in 2D
- 10.1016/j.artint.2018.07.007 may be a valid DOI for title: Explanation in Artificial Intelligence: Insights from the Social Sciences
- 10.1037/0022-3514.54.4.687 may be a valid DOI for title: The Nature of Adolescent Competencies Predicted by Preschool Delay of Gratification.
- 10.1145/3287560.3287574 may be a valid DOI for title: Explaining Explanations in AI
- 10.1145/3351095.3372850 may be a valid DOI for title: Explaining Machine Learning Classifiers through Diverse Counterfactual Explanations
- 10.1109/cisda.2015.7208643 may be a valid DOI for title: Evaluating Model Drift in Machine Learning Algorithms
- 10.1016/s0167-7152(96)00140-x may be a valid DOI for title: Sparse Spatial Autoregressions
- 10.1145/3241036 may be a valid DOI for title: The Seven Tools of Causal Inference, with Reflections on Machine Learning
- 10.5860/choice.38-4002 may be a valid DOI for title: Economic Events, Ideas, and Policies: The 1960s and After
- 10.1016/j.jeconom.2014.08.008 may be a valid DOI for title: Estimating a Spatial Autoregressive Model with an Endogenous Spatial Weight Matrix
- 10.3386/w2966 may be a valid DOI for title: Does Monetary Policy Matter? A New Test in the Spirit of Friedman and Schwartz
- 10.1038/s42256-019-0048-x may be a valid DOI for title: Stop Explaining Black Box Machine Learning Models for High Stakes Decisions and Use Interpretable Models Instead
- 10.3386/w7469 may be a valid DOI for title: Peer Effects with Random Assignment: Results for Dartmouth Roommates
- 10.1080/01621459.2017.1395341 may be a valid DOI for title: Least Ambiguous Set-Valued Classifiers with Bounded Error Levels
- 10.1109/icdcsw.2011.20 may be a valid DOI for title: Finding a" Kneedle" in a Haystack: Detecting Knee Points in System Behavior
- 10.1086/209205 may be a valid DOI for title: Choice Based on Reasons: The Case of Attraction and Compromise Effects
- 10.21034/qr.1011 may be a valid DOI for title: Are Forecasting Models Usable for Policy Analysis?
- 10.1109/tmm.2014.2330697 may be a valid DOI for title: A Simple Method to Determine If a Music Information Retrieval System Is a “Horse”
- 10.1017/cbo9780511618031.039 may be a valid DOI for title: Libertarian Paternalism Is Not an Oxymoron
- 10.1016/0165-1765(81)90067-7 may be a valid DOI for title: Some Empirical Evidence on Dynamic Inconsistency
- 10.1086/380085 may be a valid DOI for title: Save More Tomorrow™: Using Behavioral Economics to Increase Employee Saving
- 10.1007/978-1-4613-2391-4_2 may be a valid DOI for title: The Framing of Decisions and the Psychology of Choice
- 10.1177/0956797610396225 may be a valid DOI for title: How Incidental Values from the Environment Affect Decisions about Money, Risk, and Delay
- 10.1145/3287560.3287566 may be a valid DOI for title: Actionable Recourse in Linear Classification
- 10.1037/0022-3514.79.1.66 may be a valid DOI for title: Egocentric Empathy Gaps between Owners and Buyers: Misperceptions of the Endowment Effect.
- 10.2139/ssrn.3589337 may be a valid DOI for title: Modeling Multivariate Time Series in Economics: From Auto-Regressions to Recurrent Neural Networks
- 10.2139/ssrn.3063289 may be a valid DOI for title: Counterfactual Explanations without Opening the Black Box: Automated Decisions and the GDPR
- 10.1007/bf00116900 may be a valid DOI for title: Learning in the Presence of Concept Drift and Hidden Contexts
- 10.1093/biostatistics/kxp008 may be a valid DOI for title: A Penalized Matrix Decomposition, with Applications to Sparse Principal Components and Canonical Correlation Analysis
- 10.1016/j.eswa.2007.12.020 may be a valid DOI for title: The Comparisons of Data Mining Techniques for the Predictive Accuracy of Probability of Default of Credit Card Clients
- 10.1016/s0925-2312(01)00702-0 may be a valid DOI for title: Time Series Forecasting Using a Hybrid ARIMA and Neural Network Model
- 10.1109/ssci50451.2021.9660058 may be a valid DOI for title: Evaluating Robustness of Counterfactual Explanations

INVALID DOIs

- None
```

**Comment by matbesancon** [comment]
@pat-alt can you check the DOIs above and add them to your bib file?

**Comment by matbesancon** [comment]
if necessary. Let me know once this is done

**Comment by pat-alt** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by pat-alt** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.48550/arXiv.2206.07851 is OK
- 10.48550/arXiv.1412.6980 is OK
- 10.48550/arXiv.1708.07747 is OK
- 10.1109/satml54575.2023.00036 is OK
- 10.2139/ssrn.195733 is OK
- 10.1109/deeptest52559.2021.00007 is OK
- 10.3386/w20343 is OK
- 10.1257/aer.99.1.486 is OK
- 10.1017/cbo9780511618031.014 is OK
- 10.1017/cbo9780511618031.015 is OK
- 10.1016/j.inffus.2019.12.012 is OK
- 10.7551/mitpress/3077.003.0044 is OK
- 10.1007/978-1-4419-9096-9 is OK
- 10.3386/w3487 is OK
- 10.21105/joss.02704 is OK
- 10.1016/j.techsoc.2021.101852 is OK
- 10.2139/ssrn.965818 is OK
- 10.2139/ssrn.3534495 is OK
- 10.2307/2234554 is OK
- 10.3386/w4509 is OK
- 10.1109/sp.2017.49 is OK
- 10.3386/w14032 is OK
- 10.3982/ecta10168 is OK
- 10.2139/ssrn.4090707 is OK
- 10.3386/w15617 is OK
- 10.1214/18-aoas1222 is OK
- 10.1145/3514094.3534144 is OK
- 10.1080/01621459.1999.10473858 is OK
- 10.2139/ssrn.1596425 is OK
- 10.3233/ds-200028 is OK
- 10.3390/app11167274 is OK
- 10.1037/0022-3514.37.7.1097 is OK
- 10.1086/497818 is OK
- 10.2139/ssrn.203194 is OK
- 10.1287/mnsc.2017.2908 is OK
- 10.1037/0022-3514.75.3.617 is OK
- 10.1093/qje/121.4.1283 is OK
- 10.1007/978-3-030-42553-1_3 is OK
- 10.3386/w11562 is OK
- 10.1920/wp.cem.2017.0817 is OK
- 10.1017/cbo9780511618031.006 is OK
- 10.1086/717935 is OK
- 10.1109/icmla.2011.144 is OK
- 10.1016/j.ins.2008.07.024 is OK
- 10.1145/1273496.1273541 is OK
- 10.1177/2053951720926558 is OK
- 10.1509/jmkr.48.spl.s23 is OK
- 10.1109/ICDAR.1995.598994 is OK
- 10.1002/jae.2426 is OK
- 10.21105/joss.00602 is OK
- 10.1257/aer.97.3.890 is OK
- 10.1126/science.1111709 is OK
- 10.2139/ssrn.3043404 is OK
- 10.1198/1061860032148 is OK
- 10.1109/mci.2022.3155327 is OK
- 10.1017/cbo9781139175197.009 is OK
- 10.1016/0749-5978(92)90015-y is OK
- 10.1145/3313831.3376219 is OK
- 10.4314/ajst.v5i2.15330 is OK
- 10.1007/978-3-030-93842-0_6 is OK
- 10.1081/etc-120025891 is OK
- 10.1177/0956797612450302 is OK
- 10.3386/w9736 is OK
- 10.3386/w7682 is OK
- 10.2307/2298123 is OK
- 10.2139/ssrn.2523510 is OK
- 10.20955/wp.2015.012 is OK
- 10.1007/s10618-013-0342-x is OK
- 10.1016/j.artint.2018.07.007 is OK
- 10.1037/0022-3514.54.4.687 is OK
- 10.1145/3287560.3287574 is OK
- 10.1145/3351095.3372850 is OK
- 10.1109/cisda.2015.7208643 is OK
- 10.1016/s0167-7152(96)00140-x is OK
- 10.1145/3241036 is OK
- 10.5860/choice.38-4002 is OK
- 10.1016/j.jeconom.2014.08.008 is OK
- 10.3386/w2966 is OK
- 10.1038/s42256-019-0048-x is OK
- 10.3386/w7469 is OK
- 10.1080/01621459.2017.1395341 is OK
- 10.1109/icdcsw.2011.20 is OK
- 10.1086/209205 is OK
- 10.21034/qr.1011 is OK
- 10.1109/tmm.2014.2330697 is OK
- 10.1017/cbo9780511618031.039 is OK
- 10.1016/0165-1765(81)90067-7 is OK
- 10.1086/380085 is OK
- 10.1007/978-1-4613-2391-4_2 is OK
- 10.1177/0956797610396225 is OK
- 10.1145/3287560.3287566 is OK
- 10.1037/0022-3514.79.1.66 is OK
- 10.2139/ssrn.3589337 is OK
- 10.2139/ssrn.3063289 is OK
- 10.1007/bf00116900 is OK
- 10.1093/biostatistics/kxp008 is OK
- doi.org/10.1038/s41597-020-0448-0 is OK
- 10.1016/j.eswa.2007.12.020 is OK
- 10.1016/s0925-2312(01)00702-0 is OK
- 10.1214/20-AOS1965 is OK
- 10.48550/arXiv.1810.08810 is OK
- 10.1109/ssci50451.2021.9660058 is OK
- 10.1007/s10618-022-00831-6 is OK
- 10.48550/arXiv.1912.03277 is OK
- 10.1093/pan/mpq009 is OK
- 10.48550/arXiv.1903.08689 is OK
- 10.24432/C5XW20 is OK
- 10.1145/3097983.3098039 is OK
- 10.48550/arXiv.1712.08443 is OK
- 10.18653/v1/2020.nlp4convai-1.5 is OK
- 10.48550/arXiv.1907.11692 is OK
- 10.48550/arXiv.1905.12558 is OK
- 10.48550/arXiv.1706.03662 is OK
- 10.48550/arXiv.2102.12567 is OK
- 10.48550/arXiv.1805.04829 is OK
- 10.48550/arXiv.1503.05671 is OK
- 10.48550/arXiv.2106.06137 is OK

MISSING DOIs

- 10.1093/gmo/9781561592630.article.j990050 may be a valid DOI for title: "Fidelity"
- 10.3150/22-bej1484 may be a valid DOI for title: Bayes-Optimal Prediction with Frequentist Coverage Control
- 10.1006/obhd.1996.0077 may be a valid DOI for title: The Evaluability Hypothesis: An Explanation for Preference Reversals between Joint and Separate Evaluations of Alternatives
- 10.1017/cbo9780511618031.033 may be a valid DOI for title: Music, Pandas, and Muggers: On the Affective Psychology of Value.

INVALID DOIs

- None
```

**Comment by pat-alt** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.48550/arXiv.2206.07851 is OK
- 10.48550/arXiv.1412.6980 is OK
- 10.48550/arXiv.1708.07747 is OK
- 10.1109/satml54575.2023.00036 is OK
- 10.2139/ssrn.195733 is OK
- 10.1109/deeptest52559.2021.00007 is OK
- 10.3386/w20343 is OK
- 10.1257/aer.99.1.486 is OK
- 10.1017/cbo9780511618031.014 is OK
- 10.1017/cbo9780511618031.015 is OK
- 10.1016/j.inffus.2019.12.012 is OK
- 10.7551/mitpress/3077.003.0044 is OK
- 10.1007/978-1-4419-9096-9 is OK
- 10.3386/w3487 is OK
- 10.21105/joss.02704 is OK
- 10.1016/j.techsoc.2021.101852 is OK
- 10.2139/ssrn.965818 is OK
- 10.2139/ssrn.3534495 is OK
- 10.2307/2234554 is OK
- 10.3386/w4509 is OK
- 10.1109/sp.2017.49 is OK
- 10.3386/w14032 is OK
- 10.3982/ecta10168 is OK
- 10.2139/ssrn.4090707 is OK
- 10.3386/w15617 is OK
- 10.1214/18-aoas1222 is OK
- 10.1145/3514094.3534144 is OK
- 10.1080/01621459.1999.10473858 is OK
- 10.2139/ssrn.1596425 is OK
- 10.3233/ds-200028 is OK
- 10.3390/app11167274 is OK
- 10.1037/0022-3514.37.7.1097 is OK
- 10.1086/497818 is OK
- 10.2139/ssrn.203194 is OK
- 10.1287/mnsc.2017.2908 is OK
- 10.1037/0022-3514.75.3.617 is OK
- 10.1093/qje/121.4.1283 is OK
- 10.1007/978-3-030-42553-1_3 is OK
- 10.3386/w11562 is OK
- 10.1920/wp.cem.2017.0817 is OK
- 10.1017/cbo9780511618031.006 is OK
- 10.1086/717935 is OK
- 10.1109/icmla.2011.144 is OK
- 10.1016/j.ins.2008.07.024 is OK
- 10.1145/1273496.1273541 is OK
- 10.1177/2053951720926558 is OK
- 10.1509/jmkr.48.spl.s23 is OK
- 10.1109/ICDAR.1995.598994 is OK
- 10.3150/22-bej1484 is OK
- 10.1006/obhd.1996.0077 is OK
- 10.1017/cbo9780511618031.033 is OK
- 10.1002/jae.2426 is OK
- 10.21105/joss.00602 is OK
- 10.1257/aer.97.3.890 is OK
- 10.1126/science.1111709 is OK
- 10.2139/ssrn.3043404 is OK
- 10.1198/1061860032148 is OK
- 10.1109/mci.2022.3155327 is OK
- 10.1017/cbo9781139175197.009 is OK
- 10.1016/0749-5978(92)90015-y is OK
- 10.1145/3313831.3376219 is OK
- 10.4314/ajst.v5i2.15330 is OK
- 10.1007/978-3-030-93842-0_6 is OK
- 10.1081/etc-120025891 is OK
- 10.1177/0956797612450302 is OK
- 10.3386/w9736 is OK
- 10.3386/w7682 is OK
- 10.2307/2298123 is OK
- 10.2139/ssrn.2523510 is OK
- 10.20955/wp.2015.012 is OK
- 10.1007/s10618-013-0342-x is OK
- 10.1016/j.artint.2018.07.007 is OK
- 10.1037/0022-3514.54.4.687 is OK
- 10.1145/3287560.3287574 is OK
- 10.1145/3351095.3372850 is OK
- 10.1109/cisda.2015.7208643 is OK
- 10.1016/s0167-7152(96)00140-x is OK
- 10.1145/3241036 is OK
- 10.5860/choice.38-4002 is OK
- 10.1016/j.jeconom.2014.08.008 is OK
- 10.3386/w2966 is OK
- 10.1038/s42256-019-0048-x is OK
- 10.3386/w7469 is OK
- 10.1080/01621459.2017.1395341 is OK
- 10.1109/icdcsw.2011.20 is OK
- 10.1086/209205 is OK
- 10.21034/qr.1011 is OK
- 10.1109/tmm.2014.2330697 is OK
- 10.1017/cbo9780511618031.039 is OK
- 10.1016/0165-1765(81)90067-7 is OK
- 10.1086/380085 is OK
- 10.1007/978-1-4613-2391-4_2 is OK
- 10.1177/0956797610396225 is OK
- 10.1145/3287560.3287566 is OK
- 10.1037/0022-3514.79.1.66 is OK
- 10.2139/ssrn.3589337 is OK
- 10.2139/ssrn.3063289 is OK
- 10.1007/bf00116900 is OK
- 10.1093/biostatistics/kxp008 is OK
- doi.org/10.1038/s41597-020-0448-0 is OK
- 10.1016/j.eswa.2007.12.020 is OK
- 10.1016/s0925-2312(01)00702-0 is OK
- 10.1214/20-AOS1965 is OK
- 10.48550/arXiv.1810.08810 is OK
- 10.1109/ssci50451.2021.9660058 is OK
- 10.1007/s10618-022-00831-6 is OK
- 10.48550/arXiv.1912.03277 is OK
- 10.1093/pan/mpq009 is OK
- 10.48550/arXiv.1903.08689 is OK
- 10.24432/C5XW20 is OK
- 10.1145/3097983.3098039 is OK
- 10.48550/arXiv.1712.08443 is OK
- 10.18653/v1/2020.nlp4convai-1.5 is OK
- 10.48550/arXiv.1907.11692 is OK
- 10.48550/arXiv.1905.12558 is OK
- 10.48550/arXiv.1706.03662 is OK
- 10.48550/arXiv.2102.12567 is OK
- 10.48550/arXiv.1805.04829 is OK
- 10.48550/arXiv.1503.05671 is OK
- 10.48550/arXiv.2106.06137 is OK

MISSING DOIs

- 10.1093/gmo/9781561592630.article.j990050 may be a valid DOI for title: "Fidelity"

INVALID DOIs

- None
```

**Comment by pat-alt** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00130/jcon.00130/10.21105.jcon.00130.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00130/jcon.00130/10.21105.jcon.00130.pdf) :page_facing_up: :point_left:

**Comment by pat-alt** [comment]
@matbesancon ok that's all done now. Have a good weekend 😃

**Comment by matbesancon** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @JuliaCon/jcon-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/JuliaCon/proceedings-papers/pull/69

If the paper PDF and Crossref deposit XML look good in https://github.com/JuliaCon/proceedings-papers/pull/69, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.48550/arXiv.2206.07851 is OK
- 10.48550/arXiv.1412.6980 is OK
- 10.48550/arXiv.1708.07747 is OK
- 10.1109/satml54575.2023.00036 is OK
- 10.2139/ssrn.195733 is OK
- 10.1109/deeptest52559.2021.00007 is OK
- 10.3386/w20343 is OK
- 10.1257/aer.99.1.486 is OK
- 10.1017/cbo9780511618031.014 is OK
- 10.1017/cbo9780511618031.015 is OK
- 10.1016/j.inffus.2019.12.012 is OK
- 10.7551/mitpress/3077.003.0044 is OK
- 10.1007/978-1-4419-9096-9 is OK
- 10.3386/w3487 is OK
- 10.21105/joss.02704 is OK
- 10.1016/j.techsoc.2021.101852 is OK
- 10.2139/ssrn.965818 is OK
- 10.2139/ssrn.3534495 is OK
- 10.2307/2234554 is OK
- 10.3386/w4509 is OK
- 10.1109/sp.2017.49 is OK
- 10.3386/w14032 is OK
- 10.3982/ecta10168 is OK
- 10.2139/ssrn.4090707 is OK
- 10.3386/w15617 is OK
- 10.1214/18-aoas1222 is OK
- 10.1145/3514094.3534144 is OK
- 10.1080/01621459.1999.10473858 is OK
- 10.2139/ssrn.1596425 is OK
- 10.3233/ds-200028 is OK
- 10.3390/app11167274 is OK
- 10.1037/0022-3514.37.7.1097 is OK
- 10.1086/497818 is OK
- 10.2139/ssrn.203194 is OK
- 10.1287/mnsc.2017.2908 is OK
- 10.1037/0022-3514.75.3.617 is OK
- 10.1093/qje/121.4.1283 is OK
- 10.1007/978-3-030-42553-1_3 is OK
- 10.3386/w11562 is OK
- 10.1920/wp.cem.2017.0817 is OK
- 10.1017/cbo9780511618031.006 is OK
- 10.1086/717935 is OK
- 10.1109/icmla.2011.144 is OK
- 10.1016/j.ins.2008.07.024 is OK
- 10.1145/1273496.1273541 is OK
- 10.1177/2053951720926558 is OK
- 10.1509/jmkr.48.spl.s23 is OK
- 10.1109/ICDAR.1995.598994 is OK
- 10.3150/22-bej1484 is OK
- 10.1006/obhd.1996.0077 is OK
- 10.1017/cbo9780511618031.033 is OK
- 10.1002/jae.2426 is OK
- 10.21105/joss.00602 is OK
- 10.1257/aer.97.3.890 is OK
- 10.1126/science.1111709 is OK
- 10.2139/ssrn.3043404 is OK
- 10.1198/1061860032148 is OK
- 10.1109/mci.2022.3155327 is OK
- 10.1017/cbo9781139175197.009 is OK
- 10.1016/0749-5978(92)90015-y is OK
- 10.1145/3313831.3376219 is OK
- 10.4314/ajst.v5i2.15330 is OK
- 10.1007/978-3-030-93842-0_6 is OK
- 10.1081/etc-120025891 is OK
- 10.1177/0956797612450302 is OK
- 10.3386/w9736 is OK
- 10.3386/w7682 is OK
- 10.2307/2298123 is OK
- 10.2139/ssrn.2523510 is OK
- 10.20955/wp.2015.012 is OK
- 10.1007/s10618-013-0342-x is OK
- 10.1016/j.artint.2018.07.007 is OK
- 10.1037/0022-3514.54.4.687 is OK
- 10.1145/3287560.3287574 is OK
- 10.1145/3351095.3372850 is OK
- 10.1109/cisda.2015.7208643 is OK
- 10.1016/s0167-7152(96)00140-x is OK
- 10.1145/3241036 is OK
- 10.5860/choice.38-4002 is OK
- 10.1016/j.jeconom.2014.08.008 is OK
- 10.3386/w2966 is OK
- 10.1038/s42256-019-0048-x is OK
- 10.3386/w7469 is OK
- 10.1080/01621459.2017.1395341 is OK
- 10.1109/icdcsw.2011.20 is OK
- 10.1086/209205 is OK
- 10.21034/qr.1011 is OK
- 10.1109/tmm.2014.2330697 is OK
- 10.1017/cbo9780511618031.039 is OK
- 10.1016/0165-1765(81)90067-7 is OK
- 10.1086/380085 is OK
- 10.1007/978-1-4613-2391-4_2 is OK
- 10.1177/0956797610396225 is OK
- 10.1145/3287560.3287566 is OK
- 10.1037/0022-3514.79.1.66 is OK
- 10.2139/ssrn.3589337 is OK
- 10.2139/ssrn.3063289 is OK
- 10.1007/bf00116900 is OK
- 10.1093/biostatistics/kxp008 is OK
- doi.org/10.1038/s41597-020-0448-0 is OK
- 10.1016/j.eswa.2007.12.020 is OK
- 10.1016/s0925-2312(01)00702-0 is OK
- 10.1214/20-AOS1965 is OK
- 10.48550/arXiv.1810.08810 is OK
- 10.1109/ssci50451.2021.9660058 is OK
- 10.1007/s10618-022-00831-6 is OK
- 10.48550/arXiv.1912.03277 is OK
- 10.1093/pan/mpq009 is OK
- 10.48550/arXiv.1903.08689 is OK
- 10.24432/C5XW20 is OK
- 10.1145/3097983.3098039 is OK
- 10.48550/arXiv.1712.08443 is OK
- 10.18653/v1/2020.nlp4convai-1.5 is OK
- 10.48550/arXiv.1907.11692 is OK
- 10.48550/arXiv.1905.12558 is OK
- 10.48550/arXiv.1706.03662 is OK
- 10.48550/arXiv.2102.12567 is OK
- 10.48550/arXiv.1805.04829 is OK
- 10.48550/arXiv.1503.05671 is OK
- 10.48550/arXiv.2106.06137 is OK

MISSING DOIs

- 10.1093/gmo/9781561592630.article.j990050 may be a valid DOI for title: "Fidelity"

INVALID DOIs

- None
```

**Comment by matbesancon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JCON!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/JuliaCon/proceedings-papers/pull/70
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jcon.00130](https://doi.org/10.21105/jcon.00130)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by matbesancon** [comment]
thanks everyone for submitting and reviewing :)

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00130/status.svg)](https://doi.org/10.21105/jcon.00130)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jcon.00130">
  <img src="https://proceedings.juliacon.org/papers/10.21105/jcon.00130/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://proceedings.juliacon.org/papers/10.21105/jcon.00130/status.svg
   :target: https://doi.org/10.21105/jcon.00130
```
This is how it will look in your documentation:

[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00130/status.svg)](https://doi.org/10.21105/jcon.00130)

**We need your help!**

JuliaCon Proceedings is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://forms.gle/yANMe4c9YmuW9MWcA
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-jcon

**Comment by pat-alt** [comment]
Thank you very much for all your help, very happy to see this go out 🥳
